### PR TITLE
OCPBUGS-14087: Enable HCCO to reconcile over the OperatorHub's disableAllDefaultSources object

### PR DESCRIPTION
**Pull request by jparrill**
**Which issue(s) this PR fixes**:
Fixes #[OCPBUGS-14087](https://issues.redhat.com/browse/OCPBUGS-14087)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [x] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go [hunk 1]
```diff
@@ -1286,6 +1289,16 @@ func (r *reconciler) reconcileOLM(ctx context.Context, hcp *hyperv1.HostedContro
 
 	p := olm.NewOperatorLifecycleManagerParams(hcp)
 
+	// Check if the defaultSources are disabled
+	operatorHub := &configv1.OperatorHub{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: OperatorHubObjectValue,
```
Comment (enxebre) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go:1295]: I find it confusing to name a variable that contains the name `OperatorHubObjectValue`, I think we can just inline the name here

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go [hunk 2]
```diff
@@ -1298,11 +1311,19 @@ func (r *reconciler) reconcileOLM(ctx context.Context, hcp *hyperv1.HostedContro
 
 	for _, catalog := range catalogs {
 		cs := catalog.manifest()
-		if _, err := r.CreateOrUpdate(ctx, r.client, cs, func() error {
-			catalog.reconcile(cs, p)
-			return nil
-		}); err != nil {
-			errs = append(errs, fmt.Errorf("failed to reconcile catalog source %s/%s: %w", cs.Namespace, cs.Name, err))
+		if operatorHub.Spec.DisableAllDefaultSources {
+			if err := r.client.Delete(ctx, cs); err != nil {
```
Comment (enxebre) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go:1315]: shouldn't this be driven by the market place operator?

Comment (jparrill) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go:1313]: Yes, but I think we are not deploying it with the CPO

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go [hunk 3]
```diff
@@ -1286,6 +1287,16 @@ func (r *reconciler) reconcileOLM(ctx context.Context, hcp *hyperv1.HostedContro
 
 	p := olm.NewOperatorLifecycleManagerParams(hcp)
 
+	// Check if the defaultSources are disabled
+	operatorHub := &configv1.OperatorHub{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+	}
+	if err := r.client.Get(ctx, client.ObjectKeyFromObject(operatorHub), operatorHub); err != nil {
```
Comment (enxebre) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go:1296]: what if error is not found? should we append the error too?

**Comment by enxebre** [comment]
@jparrill can you please create a ticket against marketplace operator so at least is there for future efforts on alleviating cpo adhoc logic?

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go (hunk 3 above)
Comment (jparrill) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go:1296]: If the error is not found means that the operatorHub is being deployed. I think I can append the error to the slice.

**Comment by jparrill** [comment]
> @jparrill can you please create a ticket against marketplace operator so at least is there for future efforts on alleviating cpo adhoc logic?

https://issues.redhat.com/browse/OPRUN-2950

**Pull request merged**